### PR TITLE
fixes sorting issues on home / user page

### PR DESCRIPTION
**Pull request by hamilton**
I understand we will likely be changing this around, but I found the sorting issues on the home / user page were a bit broken, so I 1) have the home page showing notebooks sorted most recent at top, and 2) order the notebooks on a user page in the same way.

### CODE REVIEW

**Comment by hamilton** [comment]
good point, will do it that way then merge

**Comment by hamilton** [comment]
@wlach migrated the sorting to the view. Let me know if this seems right.

File: src/server/pages/user-page.jsx [hunk 1]
```diff
@@ -32,7 +32,8 @@ h2 {
 class UserNotebookList extends React.Component {
   constructor(props) {
     super(props)
-    this.state = { notebookList: this.props.notebookList }
+    const { notebookList } = this.props
+    this.state = { notebookList }
```
Comment (wlach) [inline src/server/pages/user-page.jsx]: Can probably revert changes to this file entirely now.

File: server/views.py [hunk 2]
```diff
@@ -51,7 +59,7 @@ def user(request, name=None):
                 'title': v[1],
                 'last_revision': NotebookRevision.objects
                 .filter(notebook_id=v[0]).last().created.isoformat(sep=' ')
```
Comment (wlach) [inline server/views.py]: I think this is going to cause a db-lookup-per-notebook, which will be slow. I suspect you can just use the annotation above with some modification.

File: server/views.py [hunk 3]
```diff
@@ -18,15 +19,18 @@ def get_user_info_dict(user):
...
     return render(
         request, 'index.html', {
             'page_data': {
                 'userInfo': get_user_info_dict(request.user),
                 # this is horrible and will not scale
                 'notebookList': [
                     {'id': v[0], 'title': v[1], 'owner': v[2], 'avatar': v[3]}
-                    for v in
-                    Notebook.objects.values_list('id', 'title', 'owner__username', 'owner__avatar')
+                    for v in notebooks
```
Comment (wlach) [inline server/views.py]: For clarity, I think we should be able to rewrite this as:

```py
'notebookList': [
    { 'id': id, 'title': title, 'owner': owner, 'avatar': avatar } for (id, title, owner, avatar) in notebooks
]
```

File: server/views.py (hunk 2 above)
Comment (hamilton) [reply server/views.py]: Good point!

Comment (hamilton) [reply server/views.py]: fwiw I think we noted that this would be an issue way back when we put this view together

File: server/views.py (hunk 3 above)
Comment (hamilton) [reply server/views.py]: @wlach I'm going to go ahead and change this in `index` as well which is where I copied from

**Pull request merged**